nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/revision2/vls.dm

```diff
@@ -313,12 +318,26 @@
 /obj/structure/overmap/proc/handle_autonomous_targeting()
 	if(flak_battery_amount >= 1)
 		handle_flak()
-	if(!weapon_types[FIRE_MODE_AMS])
-		return FALSE
-	var/datum/ship_weapon/AMS = weapon_types[FIRE_MODE_AMS]
+
+	// Get all weapons designated as autonomous and prepare to fire 
+	var/list/automated_weapons = list()
+	if ( weapon_types )
+		for( var/item in weapon_types )
+			if ( item )
```

Comment:
This is kinda unecessary honestly, as this is a foreach, it shouldn't bring out a null, unless the item was QDELETED()